openedx/frontend-app-publisher
# feat: add Restriction Type in EditCoursePage

[PROD-4008](https://2u-internal.atlassian.net/browse/PROD-4008)
-------
This PR adds restriction_type field to CourseRunForm and makes it visible only for external courses

<img width="1095" alt="Screenshot 2024-04-22 at 10 33 54 PM" src="https://github.com/openedx/frontend-app-publisher/assets/87228907/f292c143-b350-450f-aac8-0b5a47978365">

## Testing

1. Check out https://github.com/openedx/course-discovery/pull/4331
2. Open an executive education course in publisher
3. Update its restriction_type field
4. Verify that the restriction_type is appropriately updated in the backend. This can be verified through the CourseRunRestriction model in Django admin
